index.d.ts

```diff
@@ -0,0 +1,70 @@
+declare module 'dronedeploy/datastore' {
+
+}
+
+
+
+declare module 'dronedeploy/functions-wrapper' {
+
+  import {Request, Response} from "express";
+
+  /**
+   * The function callback is passed into every function request.
+   *
+   * If an error is present, then it has already been sent to the client so the response
+   * object cannot be mutated.
+   */
+  type FunctionCallback =  (err: Error | undefined, context: FunctionContext) => void;
+
+  /**
+   * The context passed into every function invocation.
+   */
+  interface FunctionContext {
+    token: string
+    originalToken: string
+  }
+
+  /**
+   * Allows consumers to pass in an array of CORS headers to be set by default for every
+   * function invocation.
+   */
+  interface Cors {
+    headers?: string[],
+  }
+
+  /**
+   * Configuration for the function
+   */
+  interface Config {
+
+    /**
+     * Indicates that authorization is required for the function and will return 401 if a valid
+     * DroneDeploy JWT token is not in the request headers.
+     */
+    authRequired: boolean;
+
+    /**
+     * Indicates whether to mock the token; useful for development purposes
+     */
+    mockToken?: string,
+
+    /**
+     * Configuration for CORS headers
+     */
+    cors?: Cors
```

Comment:
So if you `cat` the current `app-examples/IFTTT/functions/webhook/config.json` which gets passed to the `bootstrap` function you get:
```
{
  "authRequired": true,
  "mockToken": false,
  "config": {
    "cors": {
      "headers": []
    }
  }
}
```
so I think this piece still needs another iteration.